client/state/login/selectors.js

```diff
@@ -70,21 +62,15 @@ export const getTwoFactorPushPollInProgress = state => get( state, 'login.twoFac
 export const getTwoFactorPushPollSuccess = state => get( state, 'login.twoFactorAuthPushPoll.success', false );
 
 /**
- * True if two factor authentication is enabled for the logging in user.
- * False if not, null if the information is not available yet.
+ * Determines whether two factor authentication is enabled for the logging in user.
  *
  * @param  {Object}   state  Global state tree
- * @return {?Boolean}        Whether 2FA is enabled
+ * @return {Boolean}        Whether 2FA is enabled
  */
 export const isTwoFactorEnabled = ( state ) => {
 	const twoFactorAuth = get( state, 'login.twoFactorAuth' );
-	if ( ! twoFactorAuth ) {
-		return null;
-	}
 
-	return (
-		twoFactorAuth.user_id !== ''
-	);
+	return ! isEmpty( twoFactorAuth );
```

Comment:
Yes, that was my reasoning. I don't really see the point with checking that specific property if we only populate `login.twoFactorAuth` when some 2FA data was returned by the login endpoint.